### ISSUE TITLE
Reduce UI tests CI on PRs to one or two browsers, the rest on stable branches

### ISSUE THREAD

**Issue by PVince81** [owncloud/core#29501]
CI / Travis currently has way too many browsers to test and fails very often due to time outs or API rate limits. I don't think there is any real benefit of having so many browsers being tested in every PR.

Maybe we could reduce it to two browsers: Firefox (or Chrome) and Internet Explorer (for the random crap that might happen there). Run everything else on the stable branches.

This would reduce the frustration of random failures. We already have enough random CI failures to stomach with Jenkins.

@DeepDiver1975 @phil-davis @SergioBertolinSG

**Comment by phil-davis** [comment]
We already only test on chrome latest with each PR. The other browsers are only done for the (daily) Travis cron job.
Actually the tests on chrome are quite stable now.

The issue that we have is the Travis and SauceLabs job limit, which are both 5 parallel jobs (good). But when a running job is canceled in Travis, the selenium/browser session on SauceLabs is not cleaned up (Travis seems to just kill the job, with no chance for the job to send a shutdown to the SauceLabs selenium server). Then if the next Travis job in the queue is also a UI test it will try ti start a SauceLabs/selenium session. That fails because SauceLabs already has 5 sessions running. So then the next Travis job starts and get the same error... You see these as a string of Travis job fails with "!" status and run time of 2 or 3 minutes. After about 10 minutes the "orphan" SauceLabs sessions time out and new SacueLabs sessions can start.

It happened for a few jobs in the Travis queue last night. The quick manual "fix" is to click the restart button on each of those Travis jobs.

**Comment by phil-davis** [comment]
The total elapsed time for the whole set of Travis jobs (UI tests and other) is about 160 minutes. Given that Travis runs 5 jobs in parallel, that means 1 PR total Travis test result comes out of the pipeline every 32 minutes. When there is a rush people submitting PRs (or rebasing...) there can easily be 10 changes queued up = 5 hours.

Then nearly every PR in master has a backport in stable10 - doubling the queue.

When PRs are merged, Travis also queues jobs for master and stable10 branch. In this case only the latest job is run - if another PR is merged before the 1st job has run then Travis cancels the 1st job and keeps only the latest in the queue. But still these jobs also delay things through the day.

**Comment by ownclouders** [comment]
Hey, this issue has been closed because the label `status/STALE` is set and there were no updates for 7 days. Feel free to reopen this issue if you deem it appropriate.

(This is an automated comment from [GitMate.io](https://gitmate.io).)

**Comment by phil-davis** [comment]
The current state is that we run the full suite of UI tests for just chrome latest with every PR.
That has around 180 minutes elapsed time of tests. With 5 jobs in parallel that means 1 PR test run completes every 36 minutes on average.

There is work in progress to try and get this stuff running on drone. Then with enough drone resources we could run more parallel jobs, reducing elapsed time for a single PR.

Another approach to reducing the test time per PR is to separate out the edge-case scenarios (testing unusual combinations of input data) and only run those nightly/weekly... on master and stable10. Just run a single combination of each scenario (where there are scenario outlines with a table of input combinations) for ones where we now have tables of combinations.

**Comment by ownclouders** [comment]
Hey, this issue has been closed because the label `status/STALE` is set and there were no updates for 7 days. Feel free to reopen this issue if you deem it appropriate.

(This is an automated comment from [GitMate.io](https://gitmate.io).)

**Comment by phil-davis** [comment]
A set of UI tests on chrome is now running on drone with each PR. That runs much faster (each suite <10 minutes) and the number of "drone workers" can be adjusted to process as many suites in parallel as needed to keep up with CI workflow.

A full set of tests is run in Travis as a nightly Travis cron job, covering Firefox, Edge and IE as well as chrome. That takes time but is currently scheduled outside of European office hours.

This issue is done.
